File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
import java.util.*;

class Solution {
    public int solution(int[] numbers, int target) {
        return recursive(numbers, target, 0, 0);
    }
    
    private static int recursive(int[] numbers, int target, int idx, int sum) { 
        if (idx==numbers.length) {
            return (sum==target) ? 1:0;
        }
        
        int result = recursive(numbers, target, idx+1, sum-numbers[idx]);
        result += recursive(numbers, target, idx+1, sum+numbers[idx]);
        
        return result;
    }
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.cpp
#include <bits/stdc++.h>

using namespace std;

vector<int> nums;
int targetValue;

int recursive(int sum, int idx);

int solution(vector<int> numbers, int target) {
    nums = numbers; // ::는 전역을 의미
    targetValue = target;
    
    return recursive(0, 0);
}

int recursive(int sum, int idx) {
    if (idx == nums.size()) { // 기저조건
        if (sum == targetValue) {
            return 1;
        }
        
        return 0;
    }
    
    int res = recursive(sum + nums[idx], idx + 1);
    res += recursive(sum - nums[idx], idx + 1);

    return res;
}